lib/Sema/DerivedConformanceRawRepresentable.cpp

```diff
@@ -344,8 +344,9 @@ static bool canSynthesizeRawRepresentable(TypeChecker &tc, Decl *parentDecl,
   auto parentDC = cast<DeclContext>(parentDecl);
   rawType       = parentDC->mapTypeIntoContext(rawType);
 
-  if (!enumDecl->getInherited().empty() &&
-      enumDecl->getInherited().front().isError())
+  auto inherited = enumDecl->getInherited();
+  if (!inherited.empty() && inherited.front().wasValidated() &&
+      inherited.front().isError())
```

Comment:
It turns out that the `wasValidated()` check is indeed necessary. If we hit this _during_ validation of the `enumDecl`, and while the raw type is already being validated (but hasn't finished yet), `isError()` will assert.